Doc/library/typing.rst

```diff
@@ -968,520 +1075,479 @@ The module defines the following classes, functions and decorators:
       standard ``__annotations__`` attribute which has the same information.
 
 
-.. class:: TypedDict(dict)
+Other concrete types
+""""""""""""""""""""
 
-   A simple typed namespace. At runtime it is equivalent to
-   a plain :class:`dict`.
+.. class:: IO
+           TextIO
+           BinaryIO
 
-   ``TypedDict`` creates a dictionary type that expects all of its
-   instances to have a certain set of keys, where each key is
-   associated with a value of a consistent type. This expectation
-   is not checked at runtime but is only enforced by type checkers.
-   Usage::
+   Generic type ``IO[AnyStr]`` and its subclasses ``TextIO(IO[str])``
+   and ``BinaryIO(IO[bytes])``
+   represent the types of I/O streams such as returned by
+   :func:`open`. These types are in the ``typing.io`` namespace.
 
-      class Point2D(TypedDict):
-          x: int
-          y: int
-          label: str
+.. class:: Pattern
+           Match
 
-      a: Point2D = {'x': 1, 'y': 2, 'label': 'good'}  # OK
-      b: Point2D = {'z': 3, 'label': 'bad'}           # Fails type check
+   These type aliases
+   correspond to the return types from :func:`re.compile` and
+   :func:`re.match`.  These types (and the corresponding functions)
+   are generic in ``AnyStr`` and can be made specific by writing
+   ``Pattern[str]``, ``Pattern[bytes]``, ``Match[str]``, or
+   ``Match[bytes]``. These types are in the ``typing.re`` namespace.
 
-      assert Point2D(x=1, y=2, label='first') == dict(x=1, y=2, label='first')
+.. class:: Text
```

Comment:
I'm not sure about this at all, but this makes me think that `AnyStr` is kind of lonely... Since we have a section on "Asynchronous programming" maybe we can also have a section on "String types" that combines `AnyStr` and `Text`? What do you think? OTOH maybe it's good they are separate because `AnyStr` is often misunderstood (e.g. in my experience with mypy users at Dropbox).

Reply:
I need to sleep over this.

Reply:
I moved `AnyStr` to section **Building generic types**  right after `TypeVar` in  and it seems the right context for it (previously it was alone in **Predefined type variable**).

I agree that `AnyStr` and `Text` should be kept apart and I'm ok with the current placement of `Text` in section **Other concrete types** that includes `TextIO` and `Pattern`.

Reply:
I'm not seeing any new commits...